java/client/src/main/java/glide/api/models/configuration/AdvancedBaseClientConfiguration.java

```diff
@@ -0,0 +1,23 @@
+/** Copyright Valkey GLIDE Project Contributors - SPDX Identifier: Apache-2.0 */
+package glide.api.models.configuration;
+
+import lombok.Getter;
+import lombok.experimental.SuperBuilder;
+
+/**
+ * Advanced configuration settings class for creating a client. Shared settings for standalone and
+ * cluster clients.
+ */
+@Getter
+@SuperBuilder
+public abstract class AdvancedBaseClientConfiguration {
+
+    /**
+     * The duration in milliseconds to wait for a TCP/TLS connection to complete. This applies both
+     * during initial client creation and any reconnections that may occur during request processing.
+     * **Note**: A high connection timeout may lead to prolonged blocking of the entire command
+     * pipeline. If the client cannot establish a connection within the specified duration, a timeout
+     * error will occur. If not explicitly set, a default value of 250 milliseconds will be used.
```

Comment:
> a timeout error will occur.
Would it be a timeout error or a connection error that internally specify the connection attempt was timed out? Please use the exception class here